virtcontainers/pkg/mock/cc_proxy_mock.go

```diff
@@ -296,17 +300,30 @@ func (proxy *CCProxyMock) serve() {
 
 // Start invokes mock proxy instance to start listening.
 func (proxy *CCProxyMock) Start() {
+	proxy.stopped = false
 	proxy.startListening()
 	go func() {
 		for {
 			proxy.serve()
+
+			proxy.Lock()
```

Comment:
Just curious - do you need the locking if you only have one reader and one writer?